session/session.go

```diff
@@ -64,16 +82,77 @@ func (s *Session) Has(key string) bool {
 	return val != nil
 }
 
+func (s *Session) Invalidate() error {
+	s.Flush()
+	return s.Migrate(true)
+}
+
+func (s *Session) Migrate(destroy ...bool) error {
```

Comment:
In what scenarios is the function used, please?

Reply:
It is used to migrate one session to different id optionally delete previous session. It is used in Invalidate and Regenerate function(extensions).

Reply:
Could you please point out a real scenario? When we want to maintain session data but need to change the session ID?

Reply:
I see, it's for security. One more thing, I think this function can be private, User doesn't need it directly.